src/main/java/com/powsybl/openloadflow/network/LfAction.java

```diff
@@ -0,0 +1,182 @@
+                }
+                break;
+            case LineConnectionAction.NAME:
+                LineConnectionAction lineConnectionAction = (LineConnectionAction) action;
+                branch = network.getBranchById(lineConnectionAction.getLineId());
+                checkBranch(branch, lineConnectionAction.getLineId());
+                if (lineConnectionAction.isOpenSide1() && lineConnectionAction.isOpenSide2()) {
+                    disabledBranch = branch;
+                } else {
+                    throw new UnsupportedOperationException("Line connection action: only open line at both sides is supported yet.");
+                }
+                break;
+            case PhaseTapChangerTapPositionAction.NAME:
+                PhaseTapChangerTapPositionAction phaseTapChangerTapPositionAction = (PhaseTapChangerTapPositionAction) action;
+                branch = network.getBranchById(phaseTapChangerTapPositionAction.getTransformerId()); // only two windings transformer for the moment.
+                checkBranch(branch, phaseTapChangerTapPositionAction.getTransformerId()); // how to check that it is really a phase tap changer?
+                if (branch.getPiModel() instanceof SimplePiModel) {
+                    throw new UnsupportedOperationException("Phase tap changer tap connection action: only one tap in the branch {" + phaseTapChangerTapPositionAction.getTransformerId() + "}");
+                } else {
+                    tapPositionChange = new TapPositionChange(branch, phaseTapChangerTapPositionAction.getValue(), phaseTapChangerTapPositionAction.isRelativeValue());
+                }
+                break;
+            default:
+                throw new UnsupportedOperationException("Unsupported action type: " + action.getType());
+        }
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public LfBranch getDisabledBranch() {
+        return disabledBranch;
+    }
+
+    public LfBranch getEnabledBranch() {
+        return enabledBranch;
+    }
+
+    public static void apply(List<LfAction> actions, LfNetwork network, LfContingency contingency) {
+        Objects.requireNonNull(actions);
+        Objects.requireNonNull(network);
+
+        // first process connectivity part of actions
+        updateConnectivity(actions, network, contingency);
+
+        // then process remaining changes of actions
+        for (LfAction action : actions) {
+            action.apply();
+        }
+    }
+
+    private static void updateConnectivity(List<LfAction> actions, LfNetwork network, LfContingency contingency) {
+        GraphConnectivity<LfBus, LfBranch> connectivity = network.getConnectivity();
+        connectivity.setMainComponentVertex(network.getSlackBus());
+
+        // re-update connectivity according to post contingency state (revert after LfContingency apply)
+        connectivity.startTemporaryChanges();
+        contingency.getDisabledBranches().forEach(connectivity::removeEdge);
```

Comment:
Do we support a bus contingency (bus/bar section or bus from a bus/breaker topology)? If one day yes, we also have to remove the disabled buses.

Reply:
yes, we should add it now to not forget it later